ast/check.go

```diff
@@ -698,13 +722,26 @@ func (d *ArgErrDetail) Lines() []string {
 	return lines
 }
 
+func (d *ArgErrDetail) nilType() bool {
+	for i := range d.Have {
+		if types.Nil(d.Have[i]) {
+			return true
+		}
+	}
+	return false
+}
+
 // UnificationErrDetail represents a type mismatch error when two **values**
 // are unified (as in x = y).
 type UnificationErrDetail struct {
```

Comment:
Comment is a bit misleading. These are not errors themselves; they're details added onto errors. I'll fix the comment to make it clearer.